gpflow/quadrature.py

```diff
@@ -44,6 +46,14 @@ def mvnquad(func, means, covs, H, Din, Dout=()):
     to leave out the item index, i.e. f actually maps (?xD)->(?x*Dout).
     :return: quadratures (N,*Dout)
     """
+    # Figure out input shape information
+    Din = ((means.shape[1] if type(means.shape) is tuple else means.shape[1].value) if Din is None
```

Comment:
Nested (a if x else b) are hard to understand, could you please rewrite this as a regular nested if statement?

Reply:
(Also, do you actually expect `means` to be a Parameter instance ? I thought it'd always be a tf.Tensor?)